File: app/src/main/java/com/example/prashant/databinding/observer/ItemViewModel.java
package com.example.prashant.databinding.observer;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.example.prashant.databinding.data.Contact;
import com.example.prashant.databinding.repo.Repository;

import java.util.List;

import javax.inject.Inject;

public class ItemViewModel extends ViewModel {
    private LiveData<List<Contact>> contacts;
    private Repository repository;

    @Inject
    public ItemViewModel(Repository repository) {
        this.repository = repository;
    }

    public LiveData<List<Contact>> getContacts() {
        return this.contacts == null ? repository.get() : contacts;
    }
}

File: app/src/main/java/com/example/prashant/databinding/observer/ItemViewModel.kt
package com.example.prashant.databinding.observer

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.example.prashant.databinding.data.Contact
import com.example.prashant.databinding.repo.Repository

import javax.inject.Inject

class ItemViewModel @Inject constructor(
        val repository: Repository
) : ViewModel() {
    var contacts = MutableLiveData<List<Contact>>()

    fun getContacts(): LiveData<List<Contact>> {
        return if (this.contacts.value == null) repository.get() else contacts
    }

    fun init() {
        this.contacts.postValue(repository.get().value)
    }
}
